src/plugins/plugin-restify.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+var shimmer = require('shimmer');
+
+var SUPPORTED_VERSIONS = '<=4.x';
+
+function unpatchRestify(restify) {
+  shimmer.unwrap(restify, 'createServer');
+}
+
+function patchRestify(restify, api) {
+  shimmer.wrap(restify, 'createServer', createServerWrap);
+
+  function createServerWrap(createServer) {
+    return function createServerTrace() {
+      var server = createServer.apply(this, arguments);
+      server.use(middleware);
+      return server;
+    };
+  }
+
+  function middleware(req, res, next) {
+    var options = {
+      // we use the path part of the url as the span name and add the full url
+      // as a label later.
+      name: req.path(),
+      url: req.url,
+      traceContext: req.header(api.constants.TRACE_CONTEXT_HEADER_NAME, null),
+      skipFrames: 3
+    };
+
+    // TODO(ofrobots): can this be moved to the inner scope near the req.end
+    // clobber?
+    var originalEnd = res.end;
+    api.runInRootSpan(options, function(transaction) {
```

Comment:
Prefer to avoid the word `transaction`. I think we settled on `rootSpan` to describe this entity.